tests/views/test_survey.py

```diff
@@ -154,3 +159,42 @@ def test_get_survey_short_name_by_id_fdi_surveys(self, mock_request):
         self.assertEqual(get_survey_short_name_by_id("AOFDI_id"), "FDI")
         self.assertEqual(get_survey_short_name_by_id("AIFDI_id"), "FDI")
         self.assertEqual(get_survey_short_name_by_id("cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87"), "BRES")
+
+    @requests_mock.mock()
+    def test_update_survey_details_success(self, mock_request):
+        changed_survey_details = {
+            "survey_ref": '222',
+            "long_name": 'New Survey Long Name',
+            "short_name": 'QBX'
+        }
+        mock_request.get(url_get_survey_list, json=survey_list)
+        mock_request.put(url_update_survey_details)
+        mock_request.get(url_get_survey_list, json=updated_survey_list)
+        response = self.app.post(f"/surveys/edit-survey-details/QBS", data=changed_survey_details,
+                                 follow_redirects=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertIn('New Survey Long Name'.encode(), response.data)
+        self.assertIn('QBX'.encode(), response.data)
+
+    @requests_mock.mock()
+    def test_update_survey_details_failure(self, mock_request):
+        changed_survey_details = {
+            "survey_ref": '222',
+            "long_name": 'New Survey Long Name',
+            "short_name": 'QBX'
+        }
+        mock_request.get(url_get_survey_list, json=survey_list)
+        mock_request.put(url_update_survey_details, status_code=500)
+        mock_request.get(url_get_survey_list, json=updated_survey_list)
+        response = self.app.post(f"/surveys/edit-survey-details/QBS", data=changed_survey_details,
+                                 follow_redirects=True)
+        self.assertEqual(response.status_code, 500)
+        self.assertIn("Server error (Error 500)".encode(), response.data)
```

Comment:
Should there be a test for what happens if a 404 is returned when the survey isn't found?